refarch-frontend/src/stores/user.ts

```diff
@@ -0,0 +1,21 @@
+import { defineStore } from "pinia";
+import { computed, ref } from "vue";
+
+import User from "@/types/User";
+
+export interface UserState {
+  user: User | null;
```

Comment:
General topic: `null` handling is kind of irregular in different places and for me union types like `MyType | null` look not very clean. Maybe there is a more TypeScript-native like way to do things?
Sometimes we have `* | null`, sometimes we have `* | undefined`. We should find a consistent solution.